serverless/deploy/github/witness_pr/README.md

```diff
@@ -0,0 +1,84 @@
+# witness_pr
+
+`witness_pr` is a GitHub Action for handling incoming PRs from _witnesses_ containing a
+co-signed log checkpoint which the witness has verified is consistent with earlier
+log checkpoints it has seen.
+
+This action would be used by a serverless log which was participating in an ecosystem
+with non-addressable witnesses.
+
+## Operation
+
+The `witness_pr` action requires a copy of the PR branch, as well as a pristine checkout of
+the log's `master` (or `main`) branch.
+Public keys for known witnesses should already be committed on the `master` branch.
+
+The `witness_pr` action:
+1. checks that there is only one modified file present in the PR: `checkpoint.witnessed`
+2. attempts to combine signatures from the PR's `checkpoint.witnessed` file with an existing
+   `checkpoint.witnessed` file in the `master` branch, and updates the file in the PR with
+   the union of the signatures.
+3. if step (2) failed (e.g. because the log published an updated `checkpoint` file which the
```

Comment:
The explanation of what could cause this path is only clear to me because I was involved in discussion of this feature workflow. One thing to consider is an overview section above which lays out the `c` and `c.w` files, with the invariant that `c >= c.w` (i.e. the witness file can never be newer). The expected workflow for witnesses is to simply follow the `c` file, verify the evolution, then submit their signature as `c.w`.

Given the above, I think that sets the stage for realizing that there is a race condition between new checkpoints being generated by the log, and the witnesses signing them and submitting them back.

Reply:
I've added an overview chunk - PTAL?

Reply:
LG, thanks.